richtextfx/src/main/java/org/fxmisc/richtext/GenericStyledArea.java

```diff
@@ -406,42 +407,78 @@ protected void invalidated() {
 
-    private final ObjectProperty<Consumer<MouseEvent>> onInsideSelectionMousePressRelease = new SimpleObjectProperty<>(e -> {
-        CharacterHit hit = hit(e.getX(), e.getY());
-        moveTo(hit.getInsertionIndex(), SelectionPolicy.CLEAR);
+    @Override public final EventHandler<MouseEvent> getOnOutsideSelectionMousePressed() { return onOutsideSelectionMousePressed.get(); }
+    @Override public final void setOnOutsideSelectionMousePressed(EventHandler<MouseEvent> handler) { onOutsideSelectionMousePressed.set( handler ); }
+    @Override public final ObjectProperty<EventHandler<MouseEvent>> onOutsideSelectionMousePressedProperty() { return onOutsideSelectionMousePressed; }
+    private final ObjectProperty<EventHandler<MouseEvent>> onOutsideSelectionMousePressed = new SimpleObjectProperty<>( e -> 
+    {
+        if ( onOutsideSelectionMousePress.get() != null )  onOutsideSelectionMousePress.get().accept(e);
+        else {
+            CharacterHit hit = hit(e.getX(), e.getY());
+            moveTo(hit.getInsertionIndex(), SelectionPolicy.CLEAR);
+        }
     });
+
+    @Deprecated private final ObjectProperty<Consumer<MouseEvent>> onInsideSelectionMousePressRelease = new SimpleObjectProperty<>( null );
     @Override public final ObjectProperty<Consumer<MouseEvent>> onInsideSelectionMousePressReleaseProperty() { return onInsideSelectionMousePressRelease; }
 
+    @Override public final EventHandler<MouseEvent> getOnInsideSelectionMousePressReleased() { return onInsideSelectionMousePressReleased.get(); }
+    @Override public final void setOnInsideSelectionMousePressReleased(EventHandler<MouseEvent> handler) { onInsideSelectionMousePressReleased.set( handler ); }
+    @Override public final ObjectProperty<EventHandler<MouseEvent>> onInsideSelectionMousePressReleasedProperty() { return onInsideSelectionMousePressReleased; }
+    private final ObjectProperty<EventHandler<MouseEvent>> onInsideSelectionMousePressReleased = new SimpleObjectProperty<>( e ->
+    {
+        if ( onInsideSelectionMousePressRelease.get() != null )  onInsideSelectionMousePressRelease.get().accept(e);
+        else {
+            CharacterHit hit = hit(e.getX(), e.getY());
+            moveTo(hit.getInsertionIndex(), SelectionPolicy.CLEAR);
+        }
+    });
+
     private final ObjectProperty<Consumer<Point2D>> onNewSelectionDrag = new SimpleObjectProperty<>(p -> {
         CharacterHit hit = hit(p.getX(), p.getY());
         moveTo(hit.getInsertionIndex(), SelectionPolicy.ADJUST);
     });
     @Override public final ObjectProperty<Consumer<Point2D>> onNewSelectionDragProperty() { return onNewSelectionDrag; }
 
-    private final ObjectProperty<Consumer<MouseEvent>> onNewSelectionDragEnd = new SimpleObjectProperty<>(e -> {
-        CharacterHit hit = hit(e.getX(), e.getY());
-        moveTo(hit.getInsertionIndex(), SelectionPolicy.ADJUST);
-    });
```

Comment:
Looks like `onNewSelectionDragEnd` was properly implemented, but it seems you deleted it somewhere along the way and then had to re-add it?

Reply:
My bad for not explaining properly. Although implemented here, this code is not used as neither `onNewSelectionDragEndProperty()` nor `getOnNewSelectionDragEnd()` are referenced from anywhere as far as I could tell.